plugins/services/src/js/structs/Framework.js

```diff
@@ -20,6 +20,14 @@ module.exports = class Framework extends Application {
     return this.get("slave_ids");
   }
 
+  getTasks() {
```

Comment:
I believe `getkey` is a better option here just because if we think let's say a Queue, or Stack (even though we are not using them) they contain a `find` where we pass a value and returns the value if exists.

I propose `getKey("tasks")` and get rid of `getTasks`, `getCompletedTasks` and if there is not a lot of code using `getNodeIDs` I would also refactor that one to start using `getKey` instead